From d61d1b2a750d4d23c9ea086b9de85f15a26b4071 Mon Sep 17 00:00:00 2001
From: Wong Hoi Sing Edison <hswong3i@gmail.com>
Date: Wed, 23 Nov 2022 21:57:24 +0800
Subject: [PATCH] ansible-galaxy - support ``resolvelib >= 0.5.3, < 0.10.0``
 (#79399)

* Upgrade `resolvelib >= 0.5.3, < 0.10.0`

https://pypi.org/project/resolvelib/0.9.0/ released on 2022-11-17:

  * https://github.com/sarugaku/resolvelib/blob/master/CHANGELOG.rst#090-2022-11-17
  * https://github.com/sarugaku/resolvelib/releases/tag/0.9.0

Signed-off-by: Wong Hoi Sing Edison <hswong3i@pantarei-design.com>
(cherry picked from commit b148fd8dd74c8599f809f71117a86577ccfb0638)
---
 changelogs/fragments/79399-resolvelib_lt_0_10_0.yml   | 2 ++
 lib/ansible/galaxy/dependency_resolution/providers.py | 4 ++--
 requirements.txt                                      | 2 +-
 test/lib/ansible_test/_data/requirements/ansible.txt  | 2 +-
 test/sanity/code-smell/docs-build.requirements.in     | 2 +-
 test/sanity/code-smell/docs-build.requirements.txt    | 2 +-
 test/sanity/code-smell/package-data.requirements.in   | 2 +-
 test/sanity/code-smell/package-data.requirements.txt  | 2 +-
 8 files changed, 10 insertions(+), 8 deletions(-)
 create mode 100644 changelogs/fragments/79399-resolvelib_lt_0_10_0.yml

diff --git a/changelogs/fragments/79399-resolvelib_lt_0_10_0.yml b/changelogs/fragments/79399-resolvelib_lt_0_10_0.yml
new file mode 100644
index 0000000000..91b7f517d3
--- /dev/null
+++ b/changelogs/fragments/79399-resolvelib_lt_0_10_0.yml
@@ -0,0 +1,2 @@
+minor_changes:
+  - ansible-galaxy - support ``resolvelib >= 0.5.3, < 0.10.0``.
diff --git a/lib/ansible/galaxy/dependency_resolution/providers.py b/lib/ansible/galaxy/dependency_resolution/providers.py
index 817a1eb227..d52006b24d 100644
--- a/lib/ansible/galaxy/dependency_resolution/providers.py
+++ b/lib/ansible/galaxy/dependency_resolution/providers.py
@@ -42,7 +42,7 @@ except ImportError:
 
 # TODO: add python requirements to ansible-test's ansible-core distribution info and remove the hardcoded lowerbound/upperbound fallback
 RESOLVELIB_LOWERBOUND = SemanticVersion("0.5.3")
-RESOLVELIB_UPPERBOUND = SemanticVersion("0.9.0")
+RESOLVELIB_UPPERBOUND = SemanticVersion("0.10.0")
 RESOLVELIB_VERSION = SemanticVersion.from_loose_version(LooseVersion(resolvelib_version))
 
 
@@ -220,7 +220,7 @@ class CollectionDependencyProviderBase(AbstractProvider):
             Mapping of identifier, list of named tuple pairs.
             The named tuples have the entries ``requirement`` and ``parent``.
 
-        resolvelib >=0.8.0, <= 0.8.1
+        resolvelib >=0.8.0, <= 0.9.0
 
         :param identifier: The value returned by ``identify()``.
 
diff --git a/requirements.txt b/requirements.txt
index 20562c3e0f..b92e9ad4fb 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -12,4 +12,4 @@ packaging
 # NOTE: Ref: https://github.com/sarugaku/resolvelib/issues/69
 # NOTE: When updating the upper bound, also update the latest version used
 # NOTE: in the ansible-galaxy-collection test suite.
-resolvelib >= 0.5.3, < 0.9.0  # dependency resolver used by ansible-galaxy
+resolvelib >= 0.5.3, < 0.10.0  # dependency resolver used by ansible-galaxy
diff --git a/test/lib/ansible_test/_data/requirements/ansible.txt b/test/lib/ansible_test/_data/requirements/ansible.txt
index 20562c3e0f..b92e9ad4fb 100644
--- a/test/lib/ansible_test/_data/requirements/ansible.txt
+++ b/test/lib/ansible_test/_data/requirements/ansible.txt
@@ -12,4 +12,4 @@ packaging
 # NOTE: Ref: https://github.com/sarugaku/resolvelib/issues/69
 # NOTE: When updating the upper bound, also update the latest version used
 # NOTE: in the ansible-galaxy-collection test suite.
-resolvelib >= 0.5.3, < 0.9.0  # dependency resolver used by ansible-galaxy
+resolvelib >= 0.5.3, < 0.10.0  # dependency resolver used by ansible-galaxy
diff --git a/test/sanity/code-smell/docs-build.requirements.in b/test/sanity/code-smell/docs-build.requirements.in
index 02c3bfc946..736450071d 100644
--- a/test/sanity/code-smell/docs-build.requirements.in
+++ b/test/sanity/code-smell/docs-build.requirements.in
@@ -1,6 +1,6 @@
 jinja2
 pyyaml
-resolvelib < 0.9.0
+resolvelib < 0.10.0
 sphinx == 4.2.0
 sphinx-notfound-page
 sphinx-ansible-theme
diff --git a/test/sanity/code-smell/docs-build.requirements.txt b/test/sanity/code-smell/docs-build.requirements.txt
index 7e30a73287..69b4e062ce 100644
--- a/test/sanity/code-smell/docs-build.requirements.txt
+++ b/test/sanity/code-smell/docs-build.requirements.txt
@@ -27,7 +27,7 @@ pyparsing==3.0.9
 pytz==2022.2.1
 PyYAML==6.0
 requests==2.28.1
-resolvelib==0.8.1
+resolvelib==0.9.0
 rstcheck==3.5.0
 semantic-version==2.10.0
 sh==1.14.3
diff --git a/test/sanity/code-smell/package-data.requirements.in b/test/sanity/code-smell/package-data.requirements.in
index 6b58f7557a..4d5e098428 100644
--- a/test/sanity/code-smell/package-data.requirements.in
+++ b/test/sanity/code-smell/package-data.requirements.in
@@ -1,7 +1,7 @@
 docutils < 0.18  # match version required by sphinx in the docs-build sanity test
 jinja2
 pyyaml  # ansible-core requirement
-resolvelib < 0.9.0
+resolvelib < 0.10.0
 rstcheck < 4  # match version used in other sanity tests
 straight.plugin
 antsibull-changelog
diff --git a/test/sanity/code-smell/package-data.requirements.txt b/test/sanity/code-smell/package-data.requirements.txt
index 94ad68fd09..ea35932ce3 100644
--- a/test/sanity/code-smell/package-data.requirements.txt
+++ b/test/sanity/code-smell/package-data.requirements.txt
@@ -6,7 +6,7 @@ MarkupSafe==2.1.1
 packaging==21.3
 pyparsing==3.0.9
 PyYAML==6.0
-resolvelib==0.8.1
+resolvelib==0.9.0
 rstcheck==3.5.0
 semantic-version==2.10.0
 straight.plugin==1.5.0
-- 
2.38.1

